code/Controllers/CMSMain.php

```diff
@@ -1761,6 +1918,13 @@ public function currentPageID()
         return $id;
     }
 
+    public function currentRecordID(): ?int
+    {
+        $id = parent::currentRecordID();
+        $this->extend('updateCurrentPageID', $id);
+        return $id;
+    }
```

Comment:
Mirrors `currentPageID` above, but uses the same extension hook.
The CMS 6 PR will update the hook name and move it to `LeftAndMain::currentRecordID()` directly instead.